swap_meet/vendor.py

```diff
@@ -1,2 +1,138 @@
+from swap_meet.item import Item
+from swap_meet.electronics import Electronics
+from swap_meet.decor import Decor
+from swap_meet.clothing import Clothing
+
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    object contains an attribute: inventory that is an empty list with an optional parameter
+    it also contains the methods: add, remove, get_by_category, swap_items, swap_first_item,
+    get_best_by_category
+    '''
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        '''
+        appends item parameter to inventory list
+        returns the last item in the inventory list
+        '''
+        self.inventory.append(item)
+        last_inventory_item = self.inventory[-1]
+
+        return last_inventory_item
+
+    def remove(self, item):
+        '''
+        removes item parameter from inventory
+        returns item if item is removed, returns False if nothing is removed
+        '''
+        for product in self.inventory:
+            if product == item:
+                self.inventory.remove(item)
+                return product
+        
+        return False
+
+    def get_by_category(self, category):
+        '''
+        appends items from inventory that match the category parameter
+        '''
+        item_list = []
+
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        
+        return item_list
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        swaps items between self and other inventories
+        '''
+        # Helpers
+        self_inventory = self.inventory 
+        other_inventory = other_vendor.inventory
+
+        if my_item not in self_inventory or their_item not in other_inventory or len(other_inventory) == 0 or len(self_inventory) == 0:
+            return False
+        
+        for item in self_inventory:
+            if item == my_item:
+                other_vendor.add(item)
+                self.remove(item)
+
+        for item in other_inventory:
+            if item == their_item:
+                self.add(item)
+                other_vendor.remove(item)
```

Comment:
These loops are not necessary here.  The check on line 59 has already verified that `my_item` is in `self_inventory` and `their_item` is in `other_inventory`.  The reference to the item in the list will be the same as any other reference to the item, so we do not have to find the item in the list in order to work with it.  When we have an object in memory, all of the variables that are pointing to that object have the same value (ie, the reference to that object), and can be used for comparison purposes interchangably.  Fantastic use of helper methods, though!
```suggestion

        other_vendor.add(my_item)
        self.remove(my_item)

        self.add(their_item)
        other_vendor.remove(their_item)
```